services/QuillLMS/app/workers/snapshots/cache_snapshot_top_x_worker.rb

```diff
@@ -41,13 +48,28 @@ def perform(cache_key, query, user_id, timeframe, school_ids, filters)
     end
 
     private def generate_payload(query, timeframe, school_ids, filters)
+      timeframe_start = DateTime.parse(timeframe['current_start'])
+      timeframe_end = DateTime.parse(timeframe['current_end'])
       filters_symbolized = filters.symbolize_keys
 
-      QUERIES[query].run(**{
-        timeframe_start: DateTime.parse(timeframe['current_start']),
-        timeframe_end: DateTime.parse(timeframe['current_end']),
-        school_ids: school_ids
-      }.merge(filters_symbolized))
+      long_process_notifier = LongProcessNotifier.new(
+        SlowQueryError.new,
+        TOO_SLOW_THRESHOLD,
+        {
+          query: query,
+          timeframe_start: timeframe_start,
+          timeframe_end: timeframe_end,
+          school_ids: school_ids
+        }.merge(filters_symbolized)
```

Comment:
Note that in Ruby 3.1, there's a fun new shortening syntax, you can do this here (this is equivalent code for the first 4 items, since it):

```ruby
{query:, timeframe_start:, timeframe_end:, school_ids:}.merge(filters_symbolized)
# or since it's only for logging
{query:, timeframe_start:, timeframe_end:, school_ids:, filters:}